pkg/operators/otel-metrics/otel-metrics_test.go

```diff
@@ -0,0 +1,206 @@
+	err = o.exporter.Collect(context.Background(), md)
+	assert.NoError(t, err)
+
+	assert.NotEmpty(t, md.ScopeMetrics)
+	for _, sm := range md.ScopeMetrics {
+		assert.NotEmpty(t, sm)
+		foundCtr := false
+		foundGauge := false
+		for _, m := range sm.Metrics {
+			if m.Name == "ctr" {
+				foundCtr = true
+				data, ok := (m.Data).(metricdata.Sum[int64])
+				assert.True(t, ok)
+				assert.Equal(t, int64(10), data.DataPoints[0].Value)
+			}
+			if m.Name == "gauge" {
+				foundGauge = true
+				data, ok := (m.Data).(metricdata.Gauge[int64])
+				assert.True(t, ok)
+				assert.Equal(t, int64(9), data.DataPoints[0].Value)
+			}
+		}
+		assert.True(t, foundCtr)
+		assert.True(t, foundGauge)
+	}
+}
+
+func TestMetricsHistogram(
+	t *testing.T,
+) {
+	o := &otelMetricsOperator{skipListen: true}
+	globalParams := apihelpers.ToParamDescs(o.GlobalParams()).ToParams()
+	globalParams.Set(ParamOtelMetricsEnabled, "true")
+	err := o.Init(globalParams)
+	assert.NoError(t, err)
+
+	require.True(t, o.initialized)
+
+	var ds datasource.DataSource
+	var value datasource.FieldAccessor
+
+	ctx, cancel := context.WithTimeout(context.TODO(), time.Second)
+	defer cancel()
+
+	expectedBuckets := []uint64{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 3, 2, 3, 0}
+
+	prepare := func(gadgetCtx operators.GadgetContext) error {
+		var err error
+		ds, err = gadgetCtx.RegisterDataSource(datasource.TypeSingle, "metrics")
+		assert.NoError(t, err)
+		ds.AddAnnotation(AnnotationMetricsExport, "true")
+		value, err = ds.AddField("duration", api.Kind_Uint32, datasource.WithAnnotations(map[string]string{
+			AnnotationMetricsType: MetricTypeHistogram,
+		}))
+		assert.NoError(t, err)
+		return nil
+	}
+	produce := func(operators.GadgetContext) error {
+		for i := range 10 {
+			data, err := ds.NewPacketSingle()
+			assert.NoError(t, err)
+			err = value.PutUint32(data, uint32((i+1)*1000))
+			assert.NoError(t, err)
+			err = ds.EmitAndRelease(data)
+			assert.NoError(t, err)
+		}
+		cancel()
+		return nil
+	}
+
+	producer := simple.New("producer",
+		simple.WithPriority(Priority-1),
+		simple.OnInit(prepare),
+		simple.OnStart(produce),
+	)
+
+	gadgetCtx := gadgetcontext.New(ctx, "", gadgetcontext.WithDataOperators(o, producer))
+
+	err = gadgetCtx.Run(api.ParamValues{
+		"operator.otel-metrics.otel-metrics-name": "metrics:metrics",
+	})
+	assert.NoError(t, err)
+
+	md := &metricdata.ResourceMetrics{}
+
+	err = o.exporter.Collect(context.Background(), md)
+	assert.NoError(t, err)
+
+	assert.NotEmpty(t, md.ScopeMetrics)
+	for _, sm := range md.ScopeMetrics {
+		assert.NotEmpty(t, sm)
+		found := false
+		for _, m := range sm.Metrics {
+			if m.Name == "duration" {
+				found = true
+				data, ok := (m.Data).(metricdata.Histogram[int64])
+				assert.True(t, ok)
+				assert.Equal(t, 1, len(data.DataPoints))
+				assert.Equal(t, len(expectedBuckets), len(data.DataPoints[0].BucketCounts))
+				for i, v := range expectedBuckets {
+					assert.Equal(t, v, data.DataPoints[0].BucketCounts[i])
+				}
```

Comment:
I think you can just use `assert.Equal(t, expectedBuckets, data.DataPoints[0].BucketCounts)`